CMakeFilters.cmake

```diff
@@ -160,15 +160,27 @@ if (HDF5_ENABLE_SZIP_SUPPORT)
     endif ()
     set(libaec_USE_STATIC_LIBS ${HDF5_USE_LIBAEC_STATIC})
     set(SZIP_FOUND FALSE)
-    find_package (SZIP NAMES ${LIBAEC_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS ${LIBAEC_SEACH_TYPE})
+    find_package (libaec CONFIG)
     if (NOT SZIP_FOUND)
-      find_package (SZIP) # Legacy find
+      find_package (SZIP NAMES ${LIBAEC_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS ${LIBAEC_SEACH_TYPE}) # Legacy find
+      if (NOT SZIP_FOUND)
+        find_package (SZIP) # Legacy find
+      endif ()
     endif ()
     set(H5_SZIP_FOUND ${SZIP_FOUND})
     if (H5_SZIP_FOUND)
       set (H5_SZIP_INCLUDE_DIR_GEN ${SZIP_INCLUDE_DIR})
       set (H5_SZIP_INCLUDE_DIRS ${H5_SZIP_INCLUDE_DIRS} ${SZIP_INCLUDE_DIR})
-      set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
+      if (TARGET libaec::sz)
```

Comment:
not always sz (like on windows)

Reply:
Should avoid using hardcodded names - always if possible

Reply:
What CMake config file is outputting something different on Windows?

The target names are created via `libaec-config.cmake` https://gitlab.dkrz.de/k202009/libaec/-/blob/master/cmake/libaec-config.cmake.in?ref_type=heads#L25

If target doesn't exist, it should fall back on prior logic for `else()` case. 

This is only way to get both static and shared support. If you use `SZIP_LIBRARIES`, it will hardcode the specific library found `.a`, `.lib`, `.dll`, `.dylib`, or `.so`

Reply:
> hdf5 only needs static compression libs

But most external distribution require shared libraries and do not allow bundling libraries.

* Alpine - only shared - https://pkgs.alpinelinux.org/contents?name=libsz&repo=community&branch=edge&arch=x86_64
* Arch Linux - only shared - https://archlinux.org/packages/extra/x86_64/libaec/
* Debian - only shared - https://packages.debian.org/sid/amd64/libsz2/filelist

Reply:
hdf5 doesn't expose the API for compression, the reason for shared - static lib is only provided for linking the static hdf5. So if there is no static hdf5, then there is not a need for providing a static compression lib.
Also, the compression libraries do not require inline build with hdf5 - they be can be built externally and just use the proper include and lib paths to pull them into hdf5

Reply:
Anyway, without modification, existing logic is broken with `libaec` and will result in 
```
CMake Error at src/CMakeLists.txt:1204 (get_target_property):
  get_target_property() called with non-existent target
```

Reply:
> So if there is no static hdf5, then there is not a need for providing a static compression lib.

In Homebrew's case, we have both. We distribute a shared and static HDF5 from single build.

User expectation here is that each would behave as expected, depending on if they use `hdf5-shared` or `hdf5-static` target (or [FindHDF5.cmake](https://cmake.org/cmake/help/latest/module/FindHDF5.html) - `HDF5_USE_STATIC_LIBRARIES=[ON|OFF]`)

---

> Also, the compression libraries do not require inline build with hdf5 - they be can be built externally and just use the proper include and lib paths to pull them into hdf5

That is what we've done in Homebrew until 1.14.5 introduced a build regression.

Issue was also reported by FreeBSD. It will impact pretty much all repositories/distros that use CMake build. Linux distros are pretty much all use autotools so won't be impacted until v2.